### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Oleksii-Fedorenko**
[DEMO LINK](https://Oleksii-Fedorenko.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by mykhailoyanush — APPROVED** [review]
Good Job!

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +15,133 @@ const goodsFromServer: string[] = [
...
+
+  start = () => {
+    this.setState({
+      visible: true,
+    });
+  };
+
+  reverseGoods = () => {
+    this.setState(state => ({
+      revers: !state.revers,
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:41]:
```suggestion
      isReversed: !state.isReversed,
```

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +15,133 @@ const goodsFromServer: string[] = [
...
+  state = {
+    goods: goodsFromServer,
+    visible: false,
+    sortBy: '',
+    revers: false,
+  };
+
+  start = () => {
+    this.setState({
+      visible: true,
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:35]:
```suggestion
      isVisible: true,
```

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +15,133 @@ const goodsFromServer: string[] = [
...
+    if (revers) {
+      copyListOfGoods.reverse();
+    }
+
+    return copyListOfGoods;
+  };
+
+  render() {
+    const { visible } = this.state;
+    const preperedGoods = this.preperGoods();
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:89]:
```suggestion
    const preparedGoods = this.prepareGoods();
```

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**